devito/propagator.py

```diff
@@ -304,21 +329,37 @@ def generate_loops(self, loop_body):
         time_loop_stencils_a = [self.convert_equality_to_cgen(x) for x in self.time_loop_stencils_a]
 
         if self.profile:
-            start_time_stmt = omp_master + [cgen.Block([cgen.Statement("gettimeofday(&start, NULL)")])]
-            end_time_stmt = omp_master + [cgen.Block(
-                [cgen.Statement("gettimeofday(&end, NULL)")] +
-                [cgen.Statement("time += ((end.tv_sec - start.tv_sec) * ",
-                                "1000000u + end.tv_usec - start.tv_usec) / 1.e6")]
-            )]
-        else:
-            start_time_stmt = []
-            end_time_stmt = []
-
-        initial_block = omp_single + ([cgen.Block(time_stepping + time_loop_stencils_b)]
-                                      if time_stepping or time_loop_stencils_b else [])
-        initial_block = initial_block + start_time_stmt
-        end_block = end_time_stmt + omp_single + ([cgen.Block(time_loop_stencils_a)]
-                                                  if time_loop_stencils_a else end_time_stmt)
```

Comment:
omp_single should have been inside the parenthesis here :
`(omp_single + [cgen.Block(time_stepping + time_loop_stencils_b)] if time_stepping or time_loop_stencils_b else [])`
to avoid empty omp sigle section breaking the compilation. Does your change fix it?